### DISCUSSION TITLE
Compute the BM25 score of a new document

### DISCUSSION THREAD

**Discussion by wuchen95** [castorini/pyserini#1131] [Q&A] [answered]
Hi, I wonder if I can compute the BM25 score between a query and a new document which is not in the collection index. I wanna use the statistical information from the collection to compute the BM25 score of this new document. 
However, the current interface takes into a 'docid' which means the document is already in the collection index. 
Maybe I could insert the document into the index and then do the above operation. However, I do not want to index all the collection (e.g., MS-MARCO doc is large) again for just adding a single document.  So is there an incremental index interface like ES? 
Or how can I better solve the problem? Thanks!

**Comment by lintool** [comment, marked as answer]
Hi @wuchen95 - currently, the feature of incremental indexing from Lucene is not exposed in Pyserini, so to compute the BM25 score of a new document, you'll have to compute it yourself from the component tf/df and other stats.
